packages/babel-preset/index.js

```diff
@@ -28,6 +28,7 @@ module.exports = function shopifyCommonPreset(
     },
     transformReactConstantElements = false,
     isWebpack5 = false,
+    legacyDecorators = false,
```

Comment:
Decoupled this from typescript as this is specifically about decorators now.

Reply:
Why? This is needed if typescript is enabled. Now anywhere typescript is true then legacyDecorators will have to be set?

Reply:
That's fair. I think @BPScott was thinking this could be more declarative. But if this is only needed for typescript then why break the api.